feat: Send editor feedback

dcalhoun opened the pull request:
Collecting feedback from users while in editor may uncover valuable
insights.

Related: 

* https://github.com/Automattic/dotcom-forge/issues/10125
* https://github.com/wordpress-mobile/WordPress-iOS/pull/23980

## To Test:
<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

1. Open the experimentla editor.
1. Tap the more button (three vertical dots) in the top-right corner.
1. Tap Send Feedback.
1. Submit feedback.
1. Verify its delivery within the customer support system.

## Regression Notes

1. Potential unintended areas of impact
    None likely with this limited addition.
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    N/A
3. What automated tests I added (or what prevented me from doing so)
    N/A

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

dcalhoun committed:
feat: Send editor feedback

Collecting feedback from users while in editor may uncover valuable
insights.

dcalhoun committed:
docs: Add release note

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
feat: Enable navigating to help screen from within the editor

In the context of displaying the send feedback form in the editor,
merely navigating backwards meant the editor would be displayed, not the
expected help screen.

dcalhoun commented on WordPress/src/main/java/org/wordpress/android/ui/main/feedbackform/FeedbackFormActivity.kt:
```diff
@@ -45,15 +46,21 @@ class FeedbackFormActivity : AppCompatActivity() {
         )
     }
 
+    private fun navigateToHelpScreen() {
+        val intent = Intent(this, HelpActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
+        startActivity(intent)
+        finish()
+    }
```
Previously, the `FeedbackFormActivity`'s "Contact Support screen" link merely navigated backwards. This approach works fine for the activity when it's presented within the _Me_ → _Help & Support_ screen, but is odd when presented within the editor. In that context, it merely navigates back to the editor. 

To address this, I replaced the backwards navigation with a more robust, direct navigation to the _Help & Support_ screen. 

@nbradbury I'm less familiar with Android's navigation stack. Is this approach problematic? Is there a better way to go about this? 

<details><summary>Feedback link</summary>
<p>

![Screenshot_20250115_141647_Jetpack Pre-Alpha](https://github.com/user-attachments/assets/8ca82f00-f057-41df-be0d-e444b18b7261)


</p>
</details>

nbradbury replied:
@dcalhoun I think this is just fine.

nbradbury commented:
@dcalhoun This works well, but I'm wondering if we need to include something that tells us the feedback was sent from the editor? Perhaps something as simple as adding "Editor feedback: " to the beginning of the feedback text?

dcalhoun commented:
> This works well, but I'm wondering if we need to include something that tells us the feedback was sent from the editor? Perhaps something as simple as adding "Editor feedback: " to the beginning of the feedback text?

Yeah, I think that is a good idea. I'll explore flagging the feedback, with a prefix or otherwise.

dcalhoun committed:
feat: Prefix editor feedback with tag

Indicate the context for feedback submissions originating from the
editor.

dcalhoun commented:
@nbradbury I added an `[Editor]` prefix for feedback submitted from within the editor in f4bde506cec4dfe231e390e86fac1f2107ab5ae1. Let me know what you think of the prefix and implementation. Thanks!

nbradbury commented:
> I added an `[Editor]` prefix for feedback submitted from within the editor

I think that's fine! I'll approve this and leave it to you to merge.

nbradbury reviewed (approved):
:shipit:
